MovieViewController.swift

```diff
@@ -0,0 +1,295 @@
+
+      let apiKey = "a07e22bc18f5cb106bfe4cc1f83ad8ed"
+      let url = NSURL(string: "https://api.themoviedb.org/3/movie/\(endpoint!)?api_key=\(apiKey)&page=\(pagenum)")
+
+      let request = URLRequest(
+        url: url! as URL,
+        cachePolicy: NSURLRequest.CachePolicy.reloadIgnoringLocalCacheData,
+        timeoutInterval: 10)
+
+      let session = URLSession(
+        configuration: URLSessionConfiguration.default,
+        delegate: nil,
+        delegateQueue: OperationQueue.main
+      )
+      MBProgressHUD.showAdded(to: self.view, animated: true)
+
+      let task: URLSessionDataTask = session.dataTask(with: request,
+       completionHandler: { (dataOrNil, response, error) in
+        if let data = dataOrNil {
+          if let responseDictionary = try! JSONSerialization.jsonObject(
+            with: data, options:[]) as? NSDictionary {
+            self.movies = responseDictionary["results"] as? [NSDictionary]
+            self.tableView.reloadData()
+            MBProgressHUD.hide(for: self.view, animated: true)
+          }
+        }
+      })
+      task.resume()
+
+
+
+
+  }
+
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    if let movies = movies {
+      return movies.count
+    } else{
+      return 0
+    }
+
+  }
+  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cell =  tableView.dequeueReusableCell(withIdentifier: "MovieCell", for: indexPath) as! MovieCell
+
+    let movie = movies?[indexPath.row]
+    let title = movie?["original_title"] as! String
+    let overview = movie?["overview"] as! String
+    let posterpath = movie?["poster_path"] as! String
+    let baseURL = "https://image.tmdb.org/t/p/w500"
+    let imageURL = NSURL(string:baseURL + posterpath)
+
+    cell.title.text = title
+    cell.title.textColor = UIColor.gray
+    cell.overview.text = overview
+    cell.overview.textColor = UIColor.white
+    cell.backgroundColor = UIColor.black
+
+    cell.title.text = title as String
+    cell.overview.text = overview as String
+    cell.imgView.setImageWith(imageURL! as URL)
+
+    return cell
+  }
+  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    tableView.deselectRow(at: indexPath, animated: true)
+  }
+  
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destinationViewController.
+        // Pass the selected object to the new view controller.
+
+      let cell = sender as! UITableViewCell
+      let indexPath = tableView.indexPath(for: cell)
+      let movie = movies?[(indexPath?.row)!]
```

Comment:
don't force unwrap.

try to get into the practice / thought-process of optionally unwrapping and then handling the case if it doesn't succeed.

e.g.
```
if let indexPath = tableView.indexPath(for: cell), let movie = movies?[indexPath.row] 
```